File: dotnet/csharp/src/InventoryItem.cs
using System;

namespace EventSourcing
{
    public class InventoryItem
    {
        public InventoryItem(Guid id, string name, int count, bool active)
        {
            Id = id;
            Name = name;
            Count = count;
            Active = active;
        }

        public Guid Id { get; }
        public string Name { get; }
        public int Count { get; }
        public bool Active { get; }

        public override string ToString() => $"{{Id = {Id}; Name = {Name}; Count = {Count}; Active = {Active};}}";
        public override bool Equals(object obj) => obj is InventoryItem item && Id.Equals(item.Id) && Name == item.Name && Count == item.Count && Active == item.Active;
        public override int GetHashCode() => HashCode.Combine(Id, Name, Count, Active);
    }
}

File: node/src/inventoryItem.js
const { Created, Stocked, Sold, Activated, Deactivated } = require('./inventoryEvent');
const { Create, Stock, Sell, Activate, Deactivate } = require('./inventoryCommand');

class InventoryItem {
    constructor(id, name, count, active) {
        this.id = id;
        this.name = name;
        this.count = count;
        this.active = active;
    }
}

module.exports = { InventoryItem };
